oak/server/grpc_event.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 The Project Oak Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OAK_GRPC_SERVER_GRPC_EVENT_H_
+#define OAK_GRPC_SERVER_GRPC_EVENT_H_
+
+#include "grpc_stream.h"
+
+namespace oak {
+namespace grpc_server {
+
+// Representation of a gRPC event.
+//
+// An event object is added to the gRPC completion queue once an event is
+// completed. Then handle() is called to invoke any subsequent dependent
+// work.
+class BaseGrpcEvent {
```

Comment:
It occurred to me that perhaps s/Event/Callback/ would be a better naming? Effectively this is becoming a continuation-passing style async API now. Anyway up to you.

Reply:
How about EventHandler? I changed it to this for now.

I agree that Event is a bit weird because it has an action and it sounds like event should just have state

Reply:
Handler sounds good, thanks!